tests/ren_pool/test_withdraw_cancellation.py

```diff
@@ -0,0 +1,30 @@
+from brownie.test import given, strategy
+from brownie import accounts
+import pytest
+import constants as C
+
+@pytest.mark.parametrize('user', accounts[0:3]) # [owner, nodeOperator, user]
+@given(
+    amount=strategy('uint256', min_value = 1, max_value = C.POOL_BOND),
+)
+def test_ren_pool_withdraw_cancellation(ren_pool, ren_token, user, amount, owner):
+    """
+    Test withdraw cancellation happy path.
+    """
+    user_init_balance = ren_token.balanceOf(user)
+
+    # Owner locks pool (could be any other user)
+    ren_token.approve(ren_pool, C.POOL_BOND, {'from': owner})
+    ren_pool.deposit(C.POOL_BOND, {'from': owner})
+
+    # The pool is locked. We can now request withdraw
+    ren_pool.requestWithdraw(amount, {'from': owner})
+
+    # Make sure the withdraw request exists
+    assert ren_pool.withdrawRequests(owner) == amount
+    
+    # Delete the withdraw request
+    ren_pool.cancelWithdrawRequest({'from': owner})
+    
+    # Make sure the withdraw request does not exist anymore
+    assert ren_pool.withdrawRequests(owner) != amount
```

Comment:
I would say `assert ren_pool.withdrawRequests(owner) != Null` or something along those lines to reflect the non existence of the withdrawRequest. Well, actually, I'm not sure the request gets destroyed or is set back to zero (?)

Reply:
It is set to zero I believe, I'll make some tests and rectify this